motionplan/constraint.go

```diff
@@ -109,17 +109,21 @@ func (c *constraintHandler) Constraints() []string {
 }
 
 // CheckConstraints will check a given input against all constraints.
-func (c *constraintHandler) CheckConstraints(cInput *ConstraintInput) (bool, float64) {
+// Return values are:
+// -- a bool representing whether all constraints passed
+// -- if passing, a score representing the distance to a non-passing state. Inf(1) if failing.
+// -- if failing, a string naming the failed constraint.
+func (c *constraintHandler) CheckConstraints(cInput *ConstraintInput) (bool, float64, string) {
 	score := 0.
 
-	for _, cFunc := range c.constraints {
+	for name, cFunc := range c.constraints {
 		pass, cScore := cFunc(cInput)
 		if !pass {
-			return false, math.Inf(1)
+			return false, math.Inf(1), name
 		}
 		score += cScore
 	}
-	return true, score
+	return true, score, ""
```

Comment:
Any time we start returning 3 things I get the feeling like there is a better way to be doing things.  But we've already talked about needing to rework the way constraints are done so maybe this is good for now considering it will likely change in the future